src/servers/trigger.ts

```diff
@@ -181,15 +214,18 @@ const listenerOnchain = ({ lnd }) => {
   subTransactions.on("chain_transaction", onchainTransactionEventHandler)
 
   subTransactions.on("error", (err) => {
-    baseLogger.info({ err }, "error subTransactions")
+    baseLogger.error({ err }, "error subTransactions")
     subTransactions.removeAllListeners()
   })
 
   const subBlocks = subscribeToBlocks({ lnd })
-  subBlocks.on("block", () => updateUsersPendingPayment({ onchainOnly: true }))
+  subBlocks.on("block", async ({ height }) => {
+    await onchainBlockEventhandler({ lnd, height })
+  })
 
   subBlocks.on("error", (err) => {
-    console.log({ err }, "error subBlocks")
+    baseLogger.error({ err }, "error subBlocks")
+    subBlocks.removeAllListeners()
```

Comment:
I'm not sure we want to remove the listener on error? it doesn't mean the app have to shutdown?

Reply:
I removed both https://github.com/GaloyMoney/galoy/pull/428/commits/20668101e2a16ba4ad949eaa1af8980a7bbc26ed